doc/development/plugins.rst

```diff
@@ -1,397 +1,398 @@
+    program.add_transform(qml.defer_measurements)
+    program.add_transform(qml.transforms.split_non_commuting)
+
+    def supports_operation(op): 
+        return getattr(op, "name", None) in operation_names
+        
+    program.add_transform(decompose, stopping_condition=supports_operation, name="my_device")
+    program.add_transform(qml.transforms.broadcast_expand)
+
+Preprocessing and validation can also exist inside the :meth:`~devices.Device.execute` method, but placing them
+in the preprocessing program has several benefits. Validation can happen earlier, leading to fewer resources
+spent before the error is raised. Users can inspect, draw, and spec out the tapes at different stages throughout
+preprocessing. This provides users a better awareness of what the device is actually executing. When device
+gradients are used, the preprocessing transforms are tracked by the machine learning interfaces. With the
+ML framework tracking the classical component of preprocessing, the device does not need to manually track the
+classical component of any decompositions or compilation. For example,
+
+>>> @qml.qnode(qml.device('reference.qubit', wires=2))
+... def circuit(x):
+...     qml.IsingXX(x, wires=(0,1))
+...     qml.CH((0,1))
+...     return qml.expval(qml.X(0))
+>>> print(qml.draw(circuit, level="device")(0.5))
+0: ─╭●──RX(0.50)─╭●────────────╭●──RY(-1.57)─┤  <Z>
+1: ─╰X───────────╰X──RY(-0.79)─╰Z──RY(0.79)──┤     
+
+Allows the user to see that both ``IsingXX`` and ``CH`` are decomposed by the device, and that
+the diagonalizing gates for ``qml.expval(qml.X(0))`` are applied.
+
+Even with these benefits, devices can still opt to
+place some transforms inside the ``execute`` method. For example, ``default.qubit`` maps wires to simulation indices
+inside ``execute`` instead of in ``preprocess``.
+
+The :meth:`~.devices.Device.execute` can assume that device preprocessing has been run on the input
+tapes, and has no obligation to re-validate the input or provide sensible error messages. In the below example,
+we see `default.qubit` erroring out when unsupported operations and unsupported measurements are 
+
+>>> op = qml.Permute([2,1,0], wires=(0,1,2))
+>>> tape = qml.tape.QuantumScript([op], [qml.probs(wires=(0,1))])
+>>> qml.device('default.qubit').execute(tape)
+MatrixUndefinedError:
+>>> tape = qml.tape.QuantumScript([], [qml.density_matrix(wires=0)], shots=50)
+>>> qml.device('default.qubit').execute(tape)
+AttributeError: 'DensityMatrixMP' object has no attribute 'process_samples'
+
+Devices may define their own transforms following the description in the ``transforms`` module,
+or can include in-built transforms such as:
+
+.. currentmodule:: pennylane.devices.preprocess
+.. toctree::
+
+    decompose
+    validate_observables
+    validate_measurements
+    validate_device_wires
+    validate_multiprocessing_workers
+    validate_adjoint_trainable_params
+    no_sampling
+
+
+.. currentmodule:: pennylane
+.. toctree::
```

Comment:
```suggestion
.. autosummary::
    :toctree: api
```
I think this was correct afterall?

Reply:
that was what the original version was, and that caused warnings too.